### PR TITLE
Workaround FirstChanceExceptionEventArgs being trimmed

### PR DESCRIPTION
**Pull request by akoeplinger**
Fixes: https://github.com/xamarin/xamarin-android/issues/6626

### CODE REVIEW

File: src/Microsoft.Android.Sdk.ILLink/PreserveLists/System.Private.CoreLib.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<linker>
+	<!-- Workaround for https://github.com/xamarin/xamarin-android/issues/6626, remove once https://github.com/dotnet/runtime/pull/68265 lands -->
+	<assembly fullname="System.Private.CoreLib">
+		<type fullname="System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs">
+			<method signature="System.Void .ctor(System.Exception)" />
```
Comment (jonathanpeppers) [inline src/Microsoft.Android.Sdk.ILLink/PreserveLists/System.Private.CoreLib.xml:6]: This looks like the right names: https://source.dot.net/#System.Private.CoreLib/FirstChanceExceptionEventArgs.cs,9

**Review by jonathanpeppers — APPROVED** [review]

File: src/Microsoft.Android.Sdk.ILLink/PreserveLists/System.Private.CoreLib.xml (hunk 1 above)
Comment (jonathanpeppers) [reply src/Microsoft.Android.Sdk.ILLink/PreserveLists/System.Private.CoreLib.xml:6]: I think it's:

https://github.com/dotnet/runtime/blob/6c7ce85e149eaaf4cc452a7008754d5c799db938/src/mono/mono/metadata/object.c#L94

**Comment by jonpryor** [comment]
Given that dotnet/runtime#68235 has been merged and backported, do we *really* need this fix?

File: src/Microsoft.Android.Sdk.ILLink/PreserveLists/System.Private.CoreLib.xml (hunk 1 above)
Comment (akoeplinger) [reply src/Microsoft.Android.Sdk.ILLink/PreserveLists/System.Private.CoreLib.xml:6]: Yes it's that object.c code from mono, nothing in the BCL references the ctor.

**Comment by akoeplinger** [comment]
@jonpryor the backport won't be merged in time for the 6.0 release that you're shipping with, so this is to fix the issue until that backport reaches you.

**Comment by jonpryor** [comment]
```markdown
[Microsoft.Android.Sdk.ILLink] Preserve FirstChanceExceptionEventArgs

Fixes: https://github.com/xamarin/xamarin-android/issues/6626

Context: https://github.com/dotnet/runtime/pull/68235

If you create a .NET 6 app whichs ubscribes to the
[`AppDomain.FirstChanceException`][0] event:

	partial class MainActivity {
	    protected override void OnCreate (Bundle? savedInstanceState)
	    {
	        AppDomain.CurrentDomain.FirstChanceException += (o, e) => {
	            Console.WriteLine ("First chance exception!");
	       };
	    }
	}

and has a way to cause an exception to be thrown:

	partial class MainActivity {
	    public override void OnBackPressed() => throw new Exception ("wee?");
	}

and is built in Release configuration with Trimming enabled,

then the app will crash when the exception is thrown (in the above
case, by launching the app and then pressing the Back button):

	E companyname.fo: * Assertion at /__w/1/s/src/mono/mono/metadata/object.c:4179, condition `ctor' not met

The cause of the crash is that the linker removes the
[`FirstChanceExceptionEventArgs(Exception)`][1] constructor.

This was fixed in dotnet/runtime#68235 by using appropriate
`[DynamicDependency]` attributes.  However, this runtime fix is
unlikely to be released before our next RC.

Add a workaround by updating
`src/Microsoft.Android.Sdk.ILLink/PreserveLists/System.Private.CoreLib.xml`
so that the linker won't remove the
`FirstChanceExceptionEventArgs(Exception)` constructor.

[0]: https://docs.microsoft.com/en-us/dotnet/api/system.appdomain.firstchanceexception?view=net-6.0
[1]: https://docs.microsoft.com/en-us/dotnet/api/system.runtime.exceptionservices.firstchanceexceptioneventargs.-ctor?view=net-6.0
```

**Pull request merged**